zenpy/lib/api.py

```diff
@@ -1055,6 +1066,57 @@ def apply(self, macro):
 
         return self._query_zendesk(self.endpoint.apply, 'result', id=macro)
 
+    @extract_id(Macro)
+    def attachments(self, macro):
+        """
+        Shows attachments for a macro.
+        Zendesk API `Reference <https://developer.zendesk.com/rest_api/docs/support/macros#create-macro-attachment>`__.
+        :param macro: Macro object or id.
+        """
+
+        return self._query_zendesk(self.endpoint.attachments, 'macro_attachment', id=macro)
+        
+    def upload_attachment(self, fp, api_object="unassociated_macro", target_name=None):
+        """
+        Upload a file to Zendesk.
+        :param fp: file object, StringIO instance, content, or file path to be
+                   uploaded
+        :param api_object: macro to associate file with. if None, it is not associated to a macro.
+        :param token: upload token for uploading multiple files
+        :param target_name: name of the file inside Zendesk
+        :return: :class:`Upload` object containing a token and other information see
+            Zendesk API `Reference <https://developer.zendesk.com/rest_api/docs/core/attachments#uploading-files>`__.
+        """
+
+        if api_object:
+            return MacroAttachmentUploadRequest(self).post(
+                fp, target_name=target_name, api_object=api_object)
+
+        else:
+            return MacroAttachmentUploadRequest(self).post(
+                fp, target_name=target_name, api_object="unassociated_macro")
+
+    def download_attachment(self, attachment, destination):
```

Comment:
TBH it would be better to remove these methods and leave it up to the applications to handle downloading the file. The idea behind Zenpy has always to keep it as a simple wrapper over the ZD API and this is adding extra functionality that is not strictly necessary.

Reply:
Any thoughts on how this should be handled if authentication is required to download?

Reply:
You should still be able to use the authenticated `requests` session object from Zenpy to make the calls.